src/gfn/gflownet/trajectory_balance.py

```diff
@@ -53,7 +53,9 @@ def loss(self, env: Env, trajectories: Trajectories) -> TT[0, float]:
             ValueError: if the loss is NaN.
         """
         del env  # unused
-        _, _, scores = self.get_trajectories_scores(trajectories)
+        _, _, scores = self.get_trajectories_scores(
+            trajectories, recalculate_all=recalculate_all
```

Comment:
I'm wondering if there's a more explicit name for `recalculate_all` -- like `recalculate_all_logprobs`?

Reply:
yes, good idea, done